java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -402,8 +402,8 @@ private void logIfDevelopmentMode(Supplier<String> messageSupplier) {
         }
     }
 
-    private com.google.api.client.http.HttpHeaders getHeadersFromSource(HttpEventRequest request, URL targetUrl) {
-        com.google.api.client.http.HttpHeaders headers = new com.google.api.client.http.HttpHeaders();
+    private void populateHeadersFromSource(HttpRequest sourceApiRequest, HttpEventRequest request, URL targetUrl) {
+        com.google.api.client.http.HttpHeaders headers = sourceApiRequest.getHeaders();
```

Comment:
That was the issue with authentication. Some endpoints returns 401, other errors, other just empty results... hard to see. 
Any header (only authentication) added when the request was created on https://github.com/Worklytics/psoxy/pull/506/files#diff-13cad7c0c6986bc926b158c0c4f6eccfd88cfc3623a4f76b37b2e94d2099c6c0L178 was removed here as a new entity was created instead of reusing them.

Reply:
BTW I can create a PR only with this if you prefer not merging the whole PR for the release @eschultink